### PR TITLE
Acls

### PR DESCRIPTION
**Pull request by chiragsanghvi**
This PR adds support for ACL's in ermrestjs as mentioned in this issue https://github.com/informatics-isi-edu/ermrestjs/issues/487. 

For now we don't handle `select` case for columns. This is something that will be handled in ermrest.

### CODE REVIEW

File: js/core.js [hunk 1]
```diff
@@ -1780,6 +1796,30 @@
...
+         *
+         * @type {Object}
+         */
+        this.rights = jsonColumn.rights;
+
+        /**
+         * Mentions whether we should hide the value for this column
+         * @type {Boolean}
+         */
+        this.isHidden = this.rights.select === false;
```
Comment (RFSH) [inline js/core.js]: It seems like you are not using this variable in the implementation of ermrestjs. It's only being used in test cases.

File: js/utilities.js [hunk 2]
```diff
@@ -1780,3 +1780,5 @@
     ];
 
     module._systemColumns = ['RID', 'RCB', 'RMB', 'RCT', 'RMT'];
+
+    module._HIDDEN_TEXT = "--hidden--";
```
Comment (RFSH) [inline js/utilities.js]: This is not used anywhere.

File: test/specs/reference/tests/06.permissions.js [hunk 3]
```diff
@@ -1,8 +1,44 @@
-var nock = require("nock");
+var ermrestImport = require(process.env.PWD + '/test/utils/ermrest-import.js');
```
Comment (RFSH) [inline test/specs/reference/tests/06.permissions.js]: I think it's easier if I ask these questions in person, and If you can walk me through these test cases. But I'm writing my questions here just for a reference:

- Why are you changing ACLs for in `afterAll` and `beforeAll` for old test cases? Aren't those test cases only working based on annotation?
- This test spec is difficult to read. Can you break it into multiple specs? or at least have a top level `describe` that differentiates the test cases that are only based on annotation, and the test cases that are based on new ACLs?
- You've used arrow functions in this file which is ES6, I have nothing against it. But we should discuss about it in the group. It's odd that one file is using ES6.

File: test/utils/ermrest-import.js [hunk 4]
```diff
@@ -43,6 +43,20 @@ exports.importSchemas = function(configFilePaths, catalogId) {
 	return defer.promise;
 };
 
+exports.importAcls = function(params) {
```
Comment (RFSH) [inline test/utils/ermrest-import.js]: I know it's obvious what this function is for, but can you add some comments to it. These utils functions are lacking documentation. Also the indentation in line 50 and 51 are off.

File: test/specs/reference/tests/06.permissions.js [hunk 5]
```diff
@@ -1,8 +1,44 @@
...
+
+    var removeCachedCatalog = (catalogId) => {
+        var server = options.ermRest.ermrestFactory.getServer(process.env.ERMREST_URL);
+        delete server.catalogs._catalogs[catalogId];
+    };
+
+    describe("For determining reference object permissions,", () => {
 
-exports.execute = function (options) {
+        var restrictedUserId = process.env.RESTRICTED_AUTH_COOKIE_ID,
```
Comment (RFSH) [inline test/specs/reference/tests/06.permissions.js]: Please comment that `RESTRICTED_AUTH_COOKIE_ID` is something that the test framework is generating (I was confused at first that I might have to set it up myself).

File: test/specs/reference/tests/06.permissions.js (hunk 3 above)
Comment (chiragsanghvi) [reply test/specs/reference/tests/06.permissions.js]: We need to change the permissions as the restrictedUser doesn't has permissions to do anything. So when we call `canCreate` the permissions are also checked with the annotations, thats the reason I had to change them. Moreover I didn't want to test it in context of admin user. Admin already has all the rights.

Secondly,  I used the arrow functions only in test-specs. These tests are run in `node.js`. We should gradually change all test specs as we make changes to them to make development easier.

File: js/core.js (hunk 1 above)
Comment (chiragsanghvi) [reply js/core.js]: I was using these variables before when I had the select=false handling. But after discussing things with Karl I removed the logic and forgot to remove these variables.

This variable will be useful when I need to remove the column during introspection.

File: js/utilities.js (hunk 2 above)
Comment (chiragsanghvi) [reply js/utilities.js]: Same reason as mentioned above for isHidden. I will get rid of this variable though.

File: test/specs/reference/tests/06.permissions.js (hunk 5 above)
Comment (chiragsanghvi) [reply test/specs/reference/tests/06.permissions.js]: Yes, I will add the comment.

File: test/utils/ermrest-import.js (hunk 4 above)
Comment (chiragsanghvi) [reply test/utils/ermrest-import.js]: They're just proxy functions. We already have documentation for them in ermrestDataUtils. Thats the reason they don't have any documentation.

**Comment by chiragsanghvi** [comment]
Made the changes as requested

**Pull request merged**